superset-frontend/src/SqlLab/components/QueryLimitSelect/QueryLimitSelect.test.tsx

```diff
@@ -106,13 +105,25 @@ describe('QueryLimitSelect', () => {
   });
 
   it('renders dropdown select', async () => {
-    const { baseElement, getByRole } = setup({}, mockStore(initialState));
+    const { baseElement, getAllByRole, getByRole } = setup(
+      { maxRow: 50000 },
+      mockStore(initialState),
+    );
     const dropdown = baseElement.getElementsByClassName(
       'ant-dropdown-trigger',
     )[0];
 
     userEvent.click(dropdown);
     await waitFor(() => expect(getByRole('menu')).toBeInTheDocument());
+
+    const expectedLabels = [10, 100, 1000, 10000, 50000].map(i =>
```

Comment:
@giftig would you also mind adding a test when `maxRow` is a factor of 10?

Reply:
Added.